tests/test_keep-alive.py

```diff
@@ -0,0 +1,69 @@
+import gc
+
+
+def test_keep_alive_argument(capture):
+    from pybind11_tests import Parent, Child
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
+        p.addChild(Child())
+        gc.collect()
+    assert capture == """
+        Allocating child.
+        Releasing child.
+    """
+    with capture:
+        del p
+        gc.collect()
+    assert capture == "Releasing parent."
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
```

Comment:
I see. If it weren't for super long captures (there's only a few of them currently; and even they can be reworded), you could do something like

```python
with assert_capture("Line"):
    # single line
with assert_capture("Allocating child.", "Releasing child."):
    #multi line, check that stripped lines are equal
with assert_capture("Line1", "Line2", unordered=True):
    # unordered multi-line check
```

I think about half, if not most of the existing capture blocks can probably be eliminated by using return values, counters, etc; and maybe then for the remaining ones it would be sufficient to use something simple like this. It's all minor stuff of course, just would be a bit nicer.